Adafruit_SSD1306.cpp

```diff
@@ -161,10 +164,11 @@ void Adafruit_SSD1306::begin(uint8_t vccstate, uint8_t i2caddr) {
     	}
     if (hwSPI){
         digitalWrite(cs, HIGH);
-        SPI.setBitOrder(MSBFIRST);
-        SPI.setClockDivider(SPI_CLOCK_DIV8);	// 72MHz / 8 = 9Mhz
-        SPI.setDataMode(0);
-        SPI.begin();	
+        spiClass->setBitOrder(MSBFIRST);
+        spiClass->setClockDivider(SPI_CLOCK_DIV8);	// 72MHz / 8 = 9Mhz
+        spiClass->setDataMode(0);
+        spiClass->begin(cs); // Passing the chip-select here just sets the pin mode. 
+                             // The peripheral won't automatically assert/deassert the pin during operation.
```

Comment:
By not passing the chip select pin here, we were mistakenly allowing the SPI driver to set up the default chip select pin, `A2`, as an output in addition to the pin that was passed in by the user.

Reply:
See https://docs.particle.io/reference/firmware/photon/#begin--3 for details